packages/jsxstyle/jsxstyle.js

```diff
@@ -27,7 +27,16 @@ function factory(displayName, defaultProps, tagName) {
     }
 
     render() {
-      const { props, style, children } = this.props;
+      const { props = {}, style, children, domRef } = this.props;
+
+      if (domRef) {
+        if (typeof this.component === 'string') {
+          props.ref = domRef;
+        } else {
+          props.domRef = domRef;
```

Comment:
In this case, the specified `component` would need to know how to deal with `domRef`, right? I’m a bit wary of passing component props props to child components. For now I think passing `domRef` as `ref` does the trick.

I know what you mean about refs that point to a components. I’ve used that feature once in my life and it was _a terrible idea_. I’m mainly going for consistency: limited magic and no surprises.

Reply:
Yeah - you'd need to have a component that accepted `domRef`.

Seems like there are 3 options:

1. Pass `domRef` as `ref` to DOM components.  Ignore it for custom components;
2. Pass `domRef` as `ref` to DOM components or as `domRef` to custom components; or,
3. Pass `innerRef` as `ref` to the component, regardless of its type.

As far as I'm concerned, the behavior of `domRef` on a component child is undefined.  It only really makes sense in terms of DOM components, so I'm fine only supporting it on them.  I'm also fine passing it through if you prefer.

It seems that any custom value for `component` would need to conform to some interface to work correctly - `style` is only a standard prop on DOM elements.  Expecting it to conform to `domRef` too, if you want a handle to the underlying DOM element, seems reasonable.

What are you thinking?

Reply:
I’m warming up to the idea of only passing `domRef` to plain components, but I need to sleep on it 💤. Second place goes to `innerRef` but really I’m not wild about the prop name.